lib/navigator.js

```diff
@@ -1,4 +1,4 @@
-module.exports = function createNavigator(channel) {
+module.exports = function createNavigator(channel, extensionId) {
```

Comment:
Adding this param in changes the way the tests work a bit. Going to have to look at it a little more and see if this is appropriate. Feedback is definitely welcome on the approach